src/promptflow-evals/promptflow/evals/evaluate/_batch_run_client/batch_run_context.py

```diff
@@ -25,10 +26,16 @@ def __enter__(self):
             os.environ[PF_FLOW_ENTRY_IN_TMP] = "true"
             os.environ[PF_FLOW_META_LOAD_IN_SUBPROCESS] = "false"
 
+            if os.environ.get(PF_BATCH_TIMEOUT_SEC) is None:
+                os.environ[PF_BATCH_TIMEOUT_SEC] = str(PF_BATCH_TIMEOUT_SEC_DEFAULT)
+
     def __exit__(self, exc_type, exc_val, exc_tb):
         if isinstance(self.client, CodeClient):
             recover_openai_api()
 
         if isinstance(self.client, ProxyClient):
             os.environ.pop(PF_FLOW_ENTRY_IN_TMP, None)
             os.environ.pop(PF_FLOW_META_LOAD_IN_SUBPROCESS, None)
+
+            if os.environ.get(PF_BATCH_TIMEOUT_SEC) == str(PF_BATCH_TIMEOUT_SEC_DEFAULT):
```

Comment:
This might not be full proof. Is it possible to pass timeout value to client that looks at it ?